### PR TITLE
chore(ci): switch to upstream action and update workflows

### PR DESCRIPTION
**Pull request by noelmiller**
BLOCKERS: 

- [x] Change `@main` to `@version_number` when action is published
- [x] All images need to successfully build
- [x] BEFORE MERGING, readd surface images to build.
- [x] THIS NEEDS TO MERGE BEFORE MERGING: https://github.com/ublue-os/bluefin/pull/1051#pullrequestreview-1966626483

### CODE REVIEW

**Comment by noelmiller** [comment]
Need to merge this PR first: https://github.com/ublue-os/bluefin/pull/1022

**Comment by abanna** [comment]
not required but useful

```docker
RUN <<EOF
echo "Hello" >> /hello
echo "World!" >> /hello
EOF
```

File: .github/workflows/build_iso.yml [hunk 1]
```diff
@@ -1,62 +1,140 @@
...
+            image_name: bluefin-dx-asus-nvidia
+        include:
+          - major_version: 38
+            is_gts_version: true
           - major_version: 39
-            image_tag: gts
+            is_gts_version: false
     steps:
+      - name: Free Disk Space (Ubuntu)
+        uses: jlumbroso/free-disk-space@v1.3.1
```
Comment (p5) [inline .github/workflows/build_iso.yml:50]: Should we use the ublue action?
https://github.com/ublue-os/remove-unwanted-software

File: .github/workflows/build_iso.yml [hunk 2]
```diff
@@ -1,62 +1,140 @@
...
   workflow_dispatch:
-  pull_request:
-    branches:
-      - main
-    paths:
-      - '.github/workflows/build_iso.yml'
-
-env:
-  IMAGE_REGISTRY: ghcr.io/${{ github.repository_owner }}
+  workflow_call:
```
Comment (p5) [inline .github/workflows/build_iso.yml:4]: IMO controlling stuff like whether to push to R2 or GHA, and the channel (stable / testing) should be controlled in the calling workflow, but this is a nit

**Review by p5 — APPROVED** [review]
Only a nit, but otherwise LGTM

File: .github/workflows/build_iso.yml (hunk 1 above)
Comment (noelmiller) [reply .github/workflows/build_iso.yml:50]: Ublue action is out of date, this one works fine.

Comment (p5) [reply .github/workflows/build_iso.yml:50]: Out of date with upstream.  LGTM

File: .github/workflows/build_iso.yml (hunk 2 above)
Comment (noelmiller) [reply .github/workflows/build_iso.yml:4]: Would be better, but we currently manually call `build_iso.yml` on it's own for uploading ISOs, so it needs to operate independently of the other workflow.

**Review by KyleGospo — APPROVED** [review]
yo this is fire

**Pull request merged**